strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/event/artifact/MavenArtifactMetadataUpdatedEventListener.java

```diff
@@ -1,20 +1,21 @@
 package org.carlspring.strongbox.event.artifact;
 
-import org.carlspring.strongbox.event.AsyncEventListener;
 import org.carlspring.strongbox.providers.io.RepositoryPath;
 import org.carlspring.strongbox.providers.layout.Maven2LayoutProvider;
 import org.carlspring.strongbox.storage.repository.Repository;
+
+import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Component;
 
 /**
  * @author Przemyslaw Fusik
  */
 @Component
-public class MavenArtifactUpdatedEventListener
+public class MavenArtifactMetadataUpdatedEventListener
         extends BaseMavenArtifactEventListener
 {
 
-    @AsyncEventListener
+    @EventListener
```

Comment:
why this was changed to sync event?

Reply:
This listener solves the `II. GroupRepository case` from https://github.com/strongbox/strongbox/issues/827#issuecomment-429534354

Reply:
FYI: It will force the group `maven-metadata.xml` to be updated if the proxied  `maven-metadata.xml` was re-fetched.

Reply:
> FYI: It will force the group maven-metadata.xml to be updated if the proxied maven-metadata.xml was re-fetched.

yes, but why it should be sync?

the goal for async was the following:
- in case of sync it affects a lot on our benchmark builds
- with async it will not affect on current request or build 

maybe I am missing something but there is no reason to make it sync within this PR

Reply:
> It will update group maven-metadata.xml right away if proxied maven-metadata.xml was updated.
right

> We need up-to-date `maven-metadata.xml` on the request time.
how this related with previous statement?

within proxied `maven-metadata.xml` request we will get proxied `maven-metadata.xml`, and this event will update group `maven-metadata.xml`

Reply:
We need up-to-date `maven-metadata.xml` **in the group repository** on the request time.

`maven-metadata.xml` is stored physically in the `group repository`

Reply:
ok, lets get back a little and look at the `GroupRepositoryProvider` code below:
```
    public RepositoryPath fetchPath(RepositoryPath repositoryPath)
        throws IOException
    {
        RepositoryPath result = resolvePathDirectlyFromGroupPathIfPossible(repositoryPath);
        if (result != null && !RepositoryFiles.hasExpired(result))
        {
            return result;
        }
        
        return resolvePathTraversal(repositoryPath);
    }
```
we can see here that this event can be triggered only when group `maven-metadata.xml` has expired or not found, and in this case it will return first match `maven-metadata.xml` from group member. So if group member was proxy then `MavenArtifactMetadataUpdatedEventListener` will be triggered, but despite this we will return re-fetched  proxy `maven-metadata.xml` and sync event will not resolve anything here, it will only affect on request time.

Another thing is that this implementation for group repositories seems wrong, because we should always return aggregated `maven-metadata.xml` for group repositories instead of returning first match from group members if there is no aggregated one.

Reply:
Where did you take mentioned code snippet from ? 

It's not from my branch https://github.com/strongbox/strongbox/blob/issue-827/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/GroupRepositoryProvider.java#L79 so we shouldn't considering it I guess.

Reply:
> Another thing is that this implementation for group repositories seems wrong, because we should always return aggregated maven-metadata.xml for group repositories instead of returning first match from group members if there is no aggregated one.

@carlspring what do you think ? It's somehow different approach than what we agreed in #827. We can do that as well.